pyproject.toml

```diff
@@ -80,6 +80,7 @@ dependencies = [
     "tiktoken",
     "openai>=1.3",
     "pydantic>=1.10,<3",
+    "docker>=7.0.0",
```

Comment:
I think this should be an optional dependency. Docker is not need for autogen to work, it is only needed if you wish to execute function calls in a Docker container. Many people don't execute functions, some people run autogen in a docker container and cannot spin of a container inside a container anyway and some people just don't want the overhead of Docker when calling functions. There should be an optional target for that, say "docker" and then if you need docker you can install it with
``` sh
pip install pyautogen[docker]
```

Reply:
moved it to its own group. I do think it should be default on but I'll leave it to #1147 since changing the dep is just moving 1 line.